flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/WindowOperator.java

```diff
@@ -419,6 +435,14 @@ public void merge(W mergeResult,
 				registerCleanupTimer(window);
 			}
 		}
+
+		// side output input event if
+		// element not handled by any window
+		// late arriving tag has been set
+		// windowAssigner is event time and current timestamp + allowed lateness no less than element timestamp
+		if(isSkippedElement && lateDataOutputTag != null && isLate(element)) {
```

Comment:
Thanks @kl0u Good catch! 

I put `isLate` there with intention to filter out `dropped events with other reasons` which I may not aware of.  lateArrivingEvents is really `late arriving` and `dropped` events.

@aljoscha If that is redundant check, we might just remove `isLate`. 
What do you think?